### PR TITLE
Feature font awesome

### PR DESCRIPTION
**Pull request by Fasse**
#53

### CODE REVIEW

**Comment by exwima** [comment]
Nur als Anmerkung: Wenn externe Fonts eingebunden werden, sollte man im geplanten Modul, dass alle DSGVO-relevanten Infos bündeln soll, einen Hinweis darauf anzeigen. Evtl. sollte man sogar eine Option vorsehen, das Einbinden externer Fonts in einer konkreten Admidio-Installation zu deaktivieren?

**Comment by ximex** [comment]
@exwima hier werden keine fonts von externen servern geladen. (Wenn ich nichts übersehen habe)
Default wird bei Admidio nichts von externen Servern geladen

File: adm_themes/modern/css/admidio.css [hunk 1]
```diff
@@ -308,6 +308,16 @@ a.btn {
     margin-right: 6px;
 }
 
+.fa-exclamation-triangle {
+    color:#dacc32;
```
Comment (ximex) [inline adm_themes/modern/css/admidio.css]: space after :

File: adm_themes/modern/css/admidio.css [hunk 2]
```diff
@@ -308,6 +308,16 @@ a.btn {
...
+}
+
+.admidio-info-icon {
+    color:#4496dd;
```
Comment (ximex) [inline adm_themes/modern/css/admidio.css]: space after :

File: adm_themes/modern/css/admidio.css [hunk 3]
```diff
@@ -312,6 +312,18 @@ a.btn {
     color:#dacc32;
 }
 
+.admidio-btn-event-attend {
+    color: green !important;
```
Comment (ximex) [inline adm_themes/modern/css/admidio.css]: use hex code for colors

File: adm_themes/modern/css/admidio.css [hunk 4]
```diff
@@ -312,6 +312,18 @@ a.btn {
...
+}
+
+.admidio-btn-event-tentative {
+    color: purple !important;
+}
+
+.admidio-btn-event-cancel {
+    color: red !important;
```
Comment (ximex) [inline adm_themes/modern/css/admidio.css]: use hex code for colors

File: adm_themes/modern/css/admidio.css [hunk 5]
```diff
@@ -312,6 +312,18 @@ a.btn {
     color:#dacc32;
 }
 
+.admidio-btn-event-attend {
+    color: green !important;
+}
+
+.admidio-btn-event-tentative {
+    color: purple !important;
```
Comment (ximex) [inline adm_themes/modern/css/admidio.css]: use hex code for colors

**Comment by ximex** [comment]
Replace:
-  `checkbox_checked.gif`=> `fa-check-square`
-  `checkbox.gif`=> `fa-square`

**Comment by ximex** [comment]
Also change all `glyphicon` to font awesome icons

File: adm_program/system/popup_message.php [hunk 6]
```diff
@@ -165,17 +165,17 @@
     <h4 class="modal-title">'.$gL10n->get('SYS_NOTE').'</h4>
 </div>
 <div class="modal-body row">
-    <div class="col-xs-2"><img style="width: 32px; height: 32px;" src="'.THEME_URL.'/icons/error_big.png" alt="Icon" /></div>
+    <div class="col-xs-2"><i class="fas fa-times-circle fa-3x" style="color: red;"></i></div>
```
[the remark this answers is not in the document]
Comment (Fasse) [reply adm_program/system/popup_message.php]: Why?

File: adm_program/languages/de.xml [hunk 7]
```diff
@@ -1183,7 +1183,7 @@
     <string name="SYS_FORMER">Ehemaliger</string>
     <string name="SYS_FORMER_PL">Ehemalige</string>
     <string name="SYS_FONT">Schriftart</string>
-    <string name="SYS_FONT_AWESOME_DESC">Hier kannst du den Namen eines Icons von der Schriftbibliothek Font Awesome angeben (z.B. fa-user). Das gewählte Icon wird dann bei dem Menüpunkt angezeigt. Eine Auswahl der möglichen Icons findest du aus der Webseite von Font Awesome.</string>
+    <string name="SYS_FONT_AWESOME_DESC">Du kannst auf unterschiedliche Arten ein Icon hinterlegen:\n1. Ein Icon des Webfonts "Font Awesome". Gehe dazu auf #VAR1#http://fontawesome.com#VAR2# und wähle ein kostenloses Icon der Sammlung "Solid" oder "Brands" aus. Als Präfix musst du noch fa- davor setzen. Wählst du also das Icon "user" aus, musst du hier "fa-user" eintragen.\n2. Eine jpg oder png Icon. Trage dazu die vollständige URL zusammen mit dem Dateinamen des Icons ein. Alternativ kannst du das Icon auch im Theme im Ordner "icons" ablegen, dann brauchst du nur den Dateinamen anzugeben.</string>
```
[the remark this answers is not in the document]
Comment (Fasse) [reply adm_program/languages/de.xml]: done

File: adm_themes/modern/css/admidio.css (hunk 1 above)
Comment (Fasse) [reply adm_themes/modern/css/admidio.css]: done

File: adm_themes/modern/css/admidio.css (hunk 2 above)
Comment (Fasse) [reply adm_themes/modern/css/admidio.css]: done

File: adm_themes/modern/css/admidio.css (hunk 4 above)
Comment (Fasse) [reply adm_themes/modern/css/admidio.css]: why?

File: adm_themes/modern/css/admidio.css (hunk 5 above)
Comment (Fasse) [reply adm_themes/modern/css/admidio.css]: why?

File: adm_themes/modern/css/admidio.css (hunk 2 above)
Comment (Fasse) [reply adm_themes/modern/css/admidio.css]: done

File: adm_themes/modern/css/admidio.css (hunk 1 above)
Comment (Fasse) [reply adm_themes/modern/css/admidio.css]: done

**Comment by Fasse** [comment]
all changes done.
I still want to use <i> for the icons because this is the prefered style of the fontawesome description and has no impact to the html code. 
The icq icon should still be used also because of backward compatibility if users still have that profile field in use.

**Comment by ximex** [comment]
Color to hex changes: Because nearly everywhere else we use hex. I prefer to don't mix up the variants.

What is with:
- Did we need the icon tooltips so often? Eg the tooltips in the Module and Admin Menu
- In the code you used sometimes string concatenation where it isn't necessary. Like `...">'.'<i class="...` instead of `..."><i class="...`

**Comment by ximex** [comment]
As i looked again: The 3 occurs in `admidio.css` and the one in `popup_message.php` are the only on except of the special keyword `transparent`

**Comment by Fasse** [comment]
- hex values are added
- we should not discuss the tooltips with this feature request
- the string concatenation is because of layout problems if you have a linefeed and than begin the new tag. Than you have a space in the layout

**Pull request merged**